central/declarativeconfig/transform/transformer.go

```diff
@@ -0,0 +1,34 @@
+package transform
+
+import (
+	"github.com/gogo/protobuf/proto"
+	"github.com/stackrox/rox/pkg/declarativeconfig"
+	"github.com/stackrox/rox/pkg/errox"
+)
+
+// Transformer transforms a declarativeconfig.Configuration to proto.Message(s).
+type Transformer interface {
+	Transform(config declarativeconfig.Configuration) ([]proto.Message, error)
```

Comment:
I suggest we modify this interface to something closer to
```
  Transform(config declarativeconfig.Configuration) (map[reflect.Type]proto.Message, error)
```
The difficulty with the current interface can come when we start to process auth provider + groups together. If we have a map where auth provider and groups are in separate buckets, we can insert auth provider first and then groups. If we have a list, we need to find auth provider within it by casting.